playbooks/roles/nginx-proxy/tasks/main.yml

```diff
@@ -0,0 +1,33 @@
+---
+- name: install passlib for htpasswd generation
+  apt:
+    name: python3-passlib
+  when: NGINX_PROXY_PASSWORD is not none
+
+- name: generate htpasswd file
+  htpasswd:
+    path: "{{ NGINX_PROXY_HTPASSWD_PATH }}"
+    username: "{{ NGINX_PROXY_USERNAME }}"
+    password: "{{ NGINX_PROXY_PASSWORD }}"
+  when: NGINX_PROXY_PASSWORD is not none
+
+- name: set up nginx to reverse-proxy
+  template:
+    src: nginx-proxy.j2
+    dest: "{{ NGINX_PROXY_CONFIG_PATH }}"
+  notify:
+    - reload nginx
```

Comment:
I haven't tested this to see if it works anyway but, should we add the handler for this in this role explicitly? It doesn't seem to exist in the files I see here.

Reply:
Roles can trigger handlers from other roles.  The "reload nginx" handler is defined in the common-server role.

Reply:
Ah, ok :)